models/courses.model.js

```diff
@@ -98,6 +103,8 @@ const courseSchema = new mongoose.Schema(
     { timestamps: true }
 );
 
+courseSchema.index({ courseId: 1, courseName: "text" });
```

Comment:
```suggestion
courseSchema.index({ _id: 1});
```

EJEMPLO:
tourSchema.index({ price: 1 });
tourSchema.index({ ratingsAverage: -1 });
tourSchema.index({ slug: 1 });
// ESTOS SON COMBINADOS, 1 es ascendente y -1 descendente. 
tourSchema.index({ price: 1, ratingsAverage: -1 });

Por favor, crea uno para _id, para status, modality, courseName y postalCode. Todos por separado.